modules/tinymce/src/themes/silver/main/ts/ui/header/InlineHeader.ts

```diff
@@ -27,11 +27,10 @@ const { ToolbarLocation, ToolbarMode } = Options;
 export const InlineHeader = (
   editor: Editor,
   targetElm: SugarElement<HTMLElement>,
-  uiComponents: RenderUiComponents,
+  uiRefs: ReadyUiReferences,
   backstage: UiFactoryBackstage,
   floatContainer: Singleton.Value<AlloyComponent>
 ): InlineHeader => {
-  const { uiMothership, outerContainer } = uiComponents;
```

Comment:
Why did we remove this? It helps keep the code smaller, etc... to deconstruct this first. I saw the same problem in WindowManager where every time you want to access anything you end up with a massive string.

- [ ] Reconsider doing this type of thing as it makes things harder to use/write

Reply:
My end goal was to have different types at each layer that only had the required parts in, but still mirrored the original structure just to avoid a whole lot of marshalling code, but I didn't do that here just because it would bloat the PR. However, I'm not sure if you'd prefer that anyway.

My main issue with RenderUiComponents is that it was just a bag of things, and wasn't structured in any meaningful way. I preferred the explicitness of defining what exactly I was using (and its context through namespacing), As soon as ReadyuiReferences was given structure, then destructuring it quickly became less clear. Also, not everything in there was a component (motherships etc.).

I can revert the change, but I would disagree with your statement that it makes it hard to read/use/write. However, that doesn't make my view right. Happy to go with those who have had to work with the code the most.

Reply:
I maybe wasn't clear enough here, or maybe I'm not understanding what you're saying as we seem to be talking about 2 separate things.

I was fine with the type changes. What I'm talking about is you do `uiRefs.mainUi.outerContainer` numerous times below. The editor team has spent time removing that in numerous places in silver previous and instead destructing early such as `const { mainUi, uiMotherships } = uiRefs;` or passing only what's needed to reduce all that duplication and the long `a.b.c.d`, etc... strings as they were not liked and slowed down development and made it harder to find what to use when you're not used to it.

Reply:
Discussed. I'll change it to max one level of reference (e.g. mainUi.outerContainer).

- [ ] change references.

Reply:
I've made it so that in most places, there is at most one level of property access.